doc/code/qml_tape.rst

```diff
@@ -0,0 +1,160 @@
+  >>> @qml.qnode(dev, interface="torch")
+  ... def circuit(p1, p2=y, **kwargs):
+  ...     qml.RX(p1, wires=0)
+  ...     qml.RY(p2[0] * p2[1], wires=0)
+  ...     qml.RX(kwargs["p3"], wires=0)
+  ...     return qml.var(qml.PauliZ(0))
+
+  When we call the QNode, we may pass the arguments by name
+  even if defined positionally; any argument not provided will
+  use the default value.
+
+  >>> res = circuit(p1=x, p3=z)
+  >>> print(res)
+  tensor(0.2327, dtype=torch.float64, grad_fn=<SelectBackward>)
+  >>> res.backward()
+  >>> print(x.grad, y.grad, z.grad)
+  tensor(0.8396) tensor([0.0289, 0.0193]) tensor(0.8387)
+
+* **Unifying all QNodes**: The tape-mode QNode merges all QNodes (including the :class:`~.JacobianQNode`
+  and the :class:`~.PassthruQNode`) into a single unified QNode, with identicaly behaviour regardless
+  of the differentiation type.
+
+  In addition, it is now possible to inspect the internal variational quantum circuit structure
+  of QNodes when using classical backpropagation (which is not support in the standard
+  :class:`~.PassthruQNode`).
+
+* **Optimizations**: Tape mode provides various optimizations, reducing pre- and post-processing
+  overhead, and reduces the number of quantum evaluations in certain cases.
+
+.. warning::
+
+    In tape-mode, the QNode does not yet have feature-parity with the standard PennyLane
+    QNode. Features currently not available in tape mode include:
+
+    * Circuit drawing and visualization
+
+    * Metric tensor computation
+
+    * The ability to automatically extract the layer structure of variational circuits
+
+    * Tape-mode QNodes cannot be used with the ``qml.qnn`` module yet
+
+
+Quantum tapes
+-------------
+
+Under the hood, tape mode is able to provide these new features by significantly overhauling
+the internal structure of the QNode. When tape mode is enabled, the QNode is no longer
+responsible for recording quantum operations, executing devices, or computing gradients---these
+tasks have been delegated to an internal object that is created by the QNode, the **Quantum tape**.
+
+In addition to be created internally by QNodes in tape mode, quantum tapes can also be created,
+nested, expanded (via :meth:`~.tape.expand`), and executed manually. Tape subclasses also provide
+additional gradient methods:
+
+.. autosummary::
+
+    ~pennylane.tape.QuantumTape
+    ~pennylane.tape.QubitParamShiftTape
+    ~pennylane.tape.CVParamShiftTape
+    ~pennylane.tape.ReversibleTape
+
+Finally, quantum tapes are fully compatible with autodifferentiating via NumPy/Autograd,
+TensorFlow, and PyTorch:
+
+.. autosummary::
+    :toctree: api
+
+    ~pennylane.tape.interfaces.tf.TFInterface
+    ~pennylane.tape.interfaces.torch.TorchInterface
+    ~pennylane.tape.interfaces.autograd.AutogradInterface
+
+For more details and examples, please see the tape documentation.
+
+.. automodapi:: pennylane.tape
+    :no-main-docstr:
+    :include-all-objects:
+    :skip: enable_tape, disable_tape

```

Comment:
These are imported top level, so show up under the `qml` namespace! If I leave off the skip here, I get Sphinx warnings, since Sphinx doesn't like code objects being repeated